src/main/java/pikabot/Parser.java

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+
+    /**
+     * Returns a Deadline commmand after parsing input from user.
+     *
+     * @param deadlineArray String array containing input string from user.
+     * @return Deadline command.
+     * @throws DeadlineException If input is invalid.
+     */
+    public static Deadline parseDeadline(String[] deadlineArray) throws DeadlineException {
+        if (deadlineArray.length == 1) {
+            throw new DeadlineException("The description of a deadline cannot be empty.");
+        } else {
+            String[] deadlineDetails = deadlineArray[1].split("/by ", 2);
+            if (deadlineDetails.length == 1) {
+                throw new DeadlineException("The description of a deadline must contain " +
+                    "a deadline in the numerical format YYYY-MM-DD");
+            } else {
+                try {
+                    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                    return new Deadline(deadlineDetails[0], LocalDate.parse(deadlineDetails[1], dtf));
+                } catch (DateTimeParseException e) {
+                    throw new DeadlineException("Invalid deadline! Deadline has to be a " +
+                        "valid date in numerical format YYYY-MM-DD");
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns an Event command after parsing input from user.
+     *
+     * @param eventArray String array containing input string from user.
+     * @return Event command.
+     * @throws EventException If input is invalid.
+     */
+    public static Event parseEvent(String[] eventArray) throws EventException {
+        if (eventArray.length == 1) {
+            throw new EventException("The description of an event cannot be empty.");
+        } else {
+            String[] eventDetails = eventArray[1].split("/at ", 2);
+            if (eventDetails.length == 1) {
+                throw new EventException("The description of an event must contain a date in " +
+                    "the numerical format YYYY-MM-DD");
+            } else {
+                try {
+                    DateTimeFormatter dtf = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+                    return new Event(eventDetails[0], LocalDate.parse(eventDetails[1], dtf));
+                } catch (DateTimeParseException e) {
+                    throw new EventException("Invalid exception! Event has to have a valid date in " +
+                        "numerical format YYYY-MM-DD");
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks whether command to search for tasks containing a keyword is valid.
+     *
+     * @param findArray String array containing input string from user.
+     * @throws FindException If no keyword is entered.
+     */
+    public static void parseFindCommand(String[] findArray) throws FindException {
+        if (findArray.length == 1) {
+            throw new FindException();
+        }
+    }
+}
```

Comment:
I like how you seperated each parsing of the respective Task into individual methods ! 👍